### PR TITLE
feat: user interface of the app

### PR DESCRIPTION
**Pull request by namannihal**

### CODE REVIEW

File: src/secur3dit/Main.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import javafx.stage.Stage;
+
+/**
+ *
+ * @author naman
+ */
+public class Main extends Application{
+    public static Stage stage = null;
+    public static void main(String[] args) {
+        // TODO code application logic here
```
Comment (viveknathani) [inline src/secur3dit/Main.java:18]: we could get rid of this comment or does this signify that something is pending? @namannihal

File: src/secur3dit/Main.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        launch(args);
+    }
+
+    @Override
+    public void start(Stage stage) throws Exception {
+       Parent root = FXMLLoader.load(getClass().getResource("ui/homepage.fxml"));
+       GraphicsDevice gd = GraphicsEnvironment.getLocalGraphicsEnvironment().getDefaultScreenDevice();
+//       int width = gd.getDisplayMode().getWidth();
```
Comment (viveknathani) [inline src/secur3dit/Main.java:26]: these comments could be removed too

File: src/secur3dit/Main.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import java.awt.GraphicsEnvironment;
+import javafx.application.Application;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.Parent;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+
+/**
+ *
+ * @author naman
```
Comment (viveknathani) [inline src/secur3dit/Main.java:13]: consider writing your full name for consistency across the project

File: src/secur3dit/ui/UifxmlController.java [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+        a.show();
+    }
+
+    @FXML
+    void horizontalMirrorImage(ActionEvent e) throws IOException {
+        if (!checkImageInput()) {
+            return;
+        }
+
+        this.output = Filters.mirror(this.output, false);
```
Comment (viveknathani) [inline src/secur3dit/ui/UifxmlController.java]: Almost every method of the filters module or the crypto module throws exceptions that need to be caught by these controllers. We need try-catch blocks wherever these methods are being called.

Comment (priyanshul-govil) [reply src/secur3dit/ui/UifxmlController.java]: > Almost every method of the filters module or the crypto module throws exceptions that need to be caught by these controllers. We need try-catch blocks wherever these methods are being called.

Crypto module errors might be handled individually, but its better if for now all errors lead to a termination with an error.

File: src/secur3dit/ui/UifxmlController.java [hunk 5]
```diff
@@ -127,11 +126,18 @@ boolean checkImageInput() {
 
     void displayMessage() {
         Alert a = new Alert(AlertType.INFORMATION);
-        a.setTitle("Secur3dit");
+        a.setTitle("S3CUR3DIT");
```
Comment (priyanshul-govil) [inline src/secur3dit/ui/UifxmlController.java]: Should be "SECUR3DIT" and not "S3..."

**Pull request merged**